src/modm/platform/adc/stm32/adc_impl.hpp.in

```diff
@@ -257,3 +257,57 @@ modm::platform::Adc{{ id }}::acknowledgeInterruptFlags(const InterruptFlag_t fla
 {
 	ADC{{ per }}->SR = ~flags.value;
 }
+
+uintptr_t
+modm::platform::Adc{{ id }}::getDataRegisterPointer()
+{
+	return (uintptr_t) &(ADC{{ per }}->DR);
+}
+
+%% if target["family"] not in ["f3"]
+void
+modm::platform::Adc{{ id }}::enableDmaMode()
+{
+	ADC{{ per }}->CR2 |= ADC_CR2_DMA;
+}
+
+void
+modm::platform::Adc{{ id }}::disableDmaMode()
+{
+	ADC{{ per }}->CR2 &= ~ADC_CR2_DMA;
+}
+
+bool
+modm::platform::Adc{{ id }}::getAdcEnabled()
+{
+	return (ADC{{ per }}->CR2 & ADC_CR2_ADON) == ADC_CR2_ADON;
+}
+
+%%endif
+%% if target["family"] not in ["f1","f3"]
+void
+modm::platform::Adc{{ id }}::enableDMASelection()
+{
+	ADC{{ per }}->CR2 |= ADC_CR2_DDS;
+}
+
+void
+modm::platform::Adc{{ id }}::disableDMASelection()
+{
+	ADC{{ per }}->CR2 &= ~ADC_CR2_DDS;
+}
+
+void
+modm::platform::Adc{{ id }}::setEndOfConversionSelection(EndOfConversionFlag endOfConversionFlag)
```

Comment:
enable/disable call maybe better?

Reply:
maybe, but this bit always enables one thing: 
If it's set to 0 -> eocs flag set after the sequence
if it's set to 1-> eocs flag set after every conversion